docs/reference/eql/eql-search-api.asciidoc

```diff
@@ -0,0 +1,173 @@
+[[eql-search-api]]
+=== EQL search API
+++++
+<titleabbrev>EQL search</titleabbrev>
+++++
+
+Returns search results for an
+https://eql.readthedocs.io/en/latest/query-guide/index.html[Event Query Language
+(EQL)] query.
+
+[source,console]
+----
+GET sec_logs/_eql/search
+{
+  "rule": """
+    sequence with maxspan=1h
+      [file where event.action=="file_create_event"] by file.path
+      [process where user.name!="SYSTEM"] by process.executable
+      [process where user.name=="SYSTEM"] by process.executable
+  """
+}
+----
+// TEST[s/^/PUT sec_logs\n/]
+
+[[eql-search-api-request]]
+==== {api-request-title}
+
+`GET <index>/_eql/search`
+
+`POST <index>/_eql/search`
+
+[[eql-search-api-prereqs]]
+==== {api-prereq-title}
+
+The EQL search API assumes each document in an index corresponds to an event.
+To work properly, each document must contain the following field archetypes:
```

Comment:
Note for reviewers:
> To work properly, each document must contain the following field archetypes:

I'm not 100% happy with the term 'archetype' here, but I felt 'type' was a bit too close to field mapping datatypes.